lighthouse-core/report/v2/renderer/details-renderer.js

```diff
@@ -134,3 +155,12 @@ DetailsRenderer.ListDetailsJSON; // eslint-disable-line no-unused-expressions
  * }}
  */
 DetailsRenderer.CardsDetailsJSON; // eslint-disable-line no-unused-expressions
+
+/**
+ * @typedef {{
+ *     type: string,
+ *     text: string,
+ *     header: (!DetailsRenderer.DetailsJSON|undefined),
```

Comment:
to avoid circular typedefs, can do `header: {type: string, text: (string|undefined)}` like `ListDetailsJSON` does with its `items`